include/lbann/weights/weights.hpp

```diff
@@ -165,6 +167,11 @@ class weights {
   El::DistData get_matrix_distribution() const;
   void set_matrix_distribution(El::DistData dist);
 
+  // -----------------------------------------------
+  // Initialize the weight class
+  // -----------------------------------------------
+  void init(lbann_comm& comm);
```

Comment:
It's not clear what the difference is between `init` and `setup`. We should call it `set_comm`.

Reply:
Good idea. Done.